server/index.js

```diff
@@ -113,7 +111,7 @@ app.post('/api/cart', (req, res, next) => {
       }
     })
     .then(result => {
-      isNaN(req.session.cardId) ? (req.session.cartId = result.cartId) : (result.cartId = req.session.cartId);
+      isNaN(req.session.cartId) ? (req.session.cartId = result.cartId) : (result.cartId = req.session.cartId);
```

Comment:
You can just assign `cartId` without checking. `req.session.cartId = result.cartId` should be fine.